src/github.com/getlantern/flashlight/util/common.go

```diff
@@ -0,0 +1,62 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"github.com/getlantern/golog"
+	"io/ioutil"
+	"os"
+	"os/user"
+)
+
+var (
+	log = golog.LoggerFor("util")
+)
+
+type logger func(arg interface{})
+
+// Based on https://github.com/getlantern/lantern-go/blob/master/src/lantern/config/config.go
+
+// determineConfigDir() determines where to load the config by checking the
+// command line and defaulting to ~/.lantern.
+func DetermineConfigDir() (string, error) {
+	usr, err := user.Current()
+	if err != nil {
+		fmt.Errorf("Error location user home directory %v", err)
+		return "", err
+	}
+	return usr.HomeDir + "/.lantern", nil
```

Comment:
Thanks! Definitely something we want to check for now. I updated that function to check for the user home directory if it doesn't already exist. I also changed that function to use path.Join() instead. If the PAC file is missing, I also create it now to synchronize it with the YAML config. Finally, I'm using https://github.com/mitchellh/go-homedir for detecting the home directory now.